grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/fragment/IdFragment.java

```diff
@@ -75,4 +93,9 @@ public double fragmentCost() {
     public boolean hasFixedFragmentCost() {
         return true;
     }
+
+    @Override
+    public boolean operatesOnEdge() {
+        return id.getValue().startsWith(Schema.PREFIX_EDGE);
```

Comment:
I don't think this is a good check. `RelationReified` id's start with `Schema.PREFIX_EDGE`. This will be true in most cases but not all.